packages/js/src/plugins/auctionHouseModule/operations/findBidsByPublicKeyField.ts

```diff
@@ -0,0 +1,87 @@
+import { Commitment, PublicKey } from '@solana/web3.js';
+import { UnreachableCaseError } from '@/errors';
+import { Metaplex } from '@/Metaplex';
+import { Operation, OperationHandler, useOperation } from '@/types';
+import { DisposableScope } from '@/utils';
+import { BidReceiptGpaBuilder } from '../gpaBuilders';
+import { AuctionHouse, Bid, toLazyBid } from '../models';
+import { AuctionHouseProgram } from '../program';
+import { toBidReceiptAccount } from '../accounts';
+
+// -----------------
+// Operation
+// -----------------
+
+const Key = 'FindBidsByPublicKeyOperation' as const;
+
+/**
+ * @group Operations
+ * @category Constructors
+ */
+export const findBidsByPublicKeyFieldOperation =
+  useOperation<FindBidsByPublicKeyFieldOperation>(Key);
+
+/**
+ * @group Operations
+ * @category Types
+ */
+export type FindBidsByPublicKeyFieldOperation = Operation<
+  typeof Key,
+  FindBidsByPublicKeyFieldInput,
+  Bid[]
+>;
+
+/**
+ * @group Operations
+ * @category Inputs
+ */
+export type FindBidsByPublicKeyFieldInput = {
+  type: 'buyer' | 'metadata';
+  auctionHouse: AuctionHouse;
+  publicKey: PublicKey;
+  commitment?: Commitment;
+};
+
+/**
+ * @group Operations
+ * @category Handlers
+ */
+export const findBidsByPublicKeyFieldOperationHandler: OperationHandler<FindBidsByPublicKeyFieldOperation> =
+  {
+    handle: async (
+      operation: FindBidsByPublicKeyFieldOperation,
+      metaplex: Metaplex,
+      scope: DisposableScope
+    ): Promise<Bid[]> => {
+      const { auctionHouse, type, publicKey, commitment } = operation.input;
+      const accounts = AuctionHouseProgram.bidAccounts(metaplex).mergeConfig({
+        commitment,
+      });
+
+      let bidQuery: BidReceiptGpaBuilder = accounts.whereAuctionHouse(
+        auctionHouse.address
+      );
+      switch (type) {
+        case 'buyer':
+          bidQuery = bidQuery.whereBuyer(publicKey);
+          break;
+        case 'metadata':
+          bidQuery = bidQuery.whereMetadata(publicKey);
+          break;
+        default:
+          throw new UnreachableCaseError(type);
+      }
+      scope.throwIfCanceled();
+
+      return await Promise.all(
+        await bidQuery.getAndMap((account) =>
+          metaplex
+            .auctionHouse()
+            .loadBid({
+              lazyBid: toLazyBid(toBidReceiptAccount(account), auctionHouse),
+            })
+            .run(scope)
+        )
+      );
```

Comment:
Yeah I don't think we should load that for them in the default operation handler. This is going to fetch JSON metadata from URI we don't control for many many bids or listings so it's going to be very inefficient.

We should return lazy models instead but allow operation handlers to return normal models.

So the output should be something like this: 

```ts
export type FindBidsByPublicKeyFieldOutput = (Bid | LazyBid)[];
```

That way, if someone wants to write an operation handler that fetches from a web2 database instead of fetching the blockchain, they can return the loaded models directly.

Reply:
Thanks, updated!